lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+  jemit.endObject(); // attributes
+  jemit.endObject(); // data
+  jemit.endObject(); // attributes
+
+  SDFGNode sdfg = dyn_cast<SDFGNode>(op->getParentOp());
+
+  StateNode srcState = sdfg.getStateBySymRef(op.src());
+  unsigned srcIdx = sdfg.getIndexOfState(srcState);
+  jemit.printKVPair("src", srcIdx);
+
+  StateNode destState = sdfg.getStateBySymRef(op.dest());
+  unsigned destIdx = sdfg.getIndexOfState(destState);
+  jemit.printKVPair("dst", destIdx);
+
+  jemit.endObject();
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// AllocOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult printScalar(AllocOp &op, JsonEmitter &jemit) {
+  jemit.startNamedObject(op.getName());
+  jemit.printKVPair("type", "Scalar");
+
+  jemit.startNamedObject("attributes");
+
+  Type element = op.getType().getElementType();
+  if (translateTypeToSDFG(element, op.getLoc(), jemit, "dtype").failed())
+    return failure();
+
+  jemit.startNamedList("shape");
+  jemit.startEntry();
+  jemit.printInt(1);
+  jemit.endList(); // shape
+
+  jemit.printKVPair("transient", "false", /*stringify=*/false);
+  jemit.printKVPair("storage", "Default");
+  jemit.printKVPair("lifetime", "Scope");
+
+  jemit.endObject(); // attributes
+
+  jemit.endObject();
+  return success();
+}
+
+LogicalResult translateAllocToSDFG(AllocOp &op, JsonEmitter &jemit) {
+  if (op.getType().getShape().size() == 0)
+    return printScalar(op, jemit);
+
+  jemit.startNamedObject(op.getName());
+  jemit.printKVPair("type", "Array");
+
+  jemit.startNamedObject("attributes");
+  // TODO: Print the values you can derive
+  // jemit.printKVPair("allow_conflicts", "false", /*stringify=*/false);
+
+  jemit.startNamedList("strides");
+  ArrayRef<int64_t> shape = op.getType().getIntegers();
+
+  for (int i = shape.size() - 1; i > 0; --i) {
+    jemit.startEntry();
+    jemit.printInt(shape[i]);
+  }
+
+  jemit.startEntry();
+  jemit.printInt(1);
+  jemit.endList(); // strides
+
+  // jemit.printKVPair("total_size", "4");
+  jemit.startNamedList("offset");
+  for (unsigned i = 0; i < shape.size(); ++i) {
+    jemit.startEntry();
+    jemit.printInt(0);
+  }
+  jemit.endList(); // offset
+
+  // jemit.printKVPair("may_alias", "false", /*stringify*/false);
+  // jemit.printKVPair("alignment", 0, /*stringify*/false);
+  Type element = op.getType().getElementType();
+  if (translateTypeToSDFG(element, op.getLoc(), jemit, "dtype").failed())
+    return failure();
+
+  jemit.startNamedList("shape");
+  for (int64_t s : shape) {
+    jemit.startEntry();
+    jemit.printInt(s);
+  }
+  jemit.endList(); // shape
+
+  jemit.printKVPair("transient", "false", /*stringify=*/false);
+  jemit.printKVPair("storage", "Default");
+  jemit.printKVPair("lifetime", "Scope");
+  // jemit.startNamedObject("location");
+  // jemit.endObject(); // location
+
+  // jemit.printKVPair("debuginfo", "null", /*stringify=*/false);
```

Comment:
you should be able to insert debuginfo from the source mlir file/line[/column] or `loc` data